### ISSUE TITLE
Schema Documentation

### ISSUE THREAD

**Issue by SafaAlfulaij** [HackSoftware/Django-Styleguide#65]
This seems like a good styleguide.
How do you handle proper schema documentation similar to how viewsets generate the schema?

```
Sample Viewset:
  id:
    retrieve
    update
    delete
  list
  create
```

As well as proper filtering and ordering options. (using a serializer for filtering is awesome idea, but schemas don't understand that)
Thanks.

**Comment by thommor** [comment]
We went down the route of having seperate API Views and tried bringing them together into a single view but as you described that does not work well with API schema generation.

We ended up defining seperate APIs (list, retrieve) etc and then bringing them together into a ViewSet. This may be abit more longwinded but maintains that full seperation but gives schema generators more chance to document your endpoints.

We currently use [drf-spectacular](https://drf-spectacular.readthedocs.io/en/latest/) which allows you to specify parameters for ordering/filtering/pagination.

Happy to share some examples if people are interested.
